### PR TITLE
(RC2) (Test only) Code cleanup (Tests)

### PR DESCRIPTION
**Pull request by ajcvickers**
Plus some test refactoring to make cleanup go smoother next time.

### CODE REVIEW

File: test/EFCore.Cosmos.FunctionalTests/EndToEndCosmosTest.cs [hunk 1]
```diff
@@ -1525,8 +1524,21 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             => modelBuilder.Entity<Customer>(
                 cb =>
                 {
-                    cb.HasPartitionKey(c => new { c.PartitionKey1, c.PartitionKey2, c.PartitionKey3 });
-                    cb.HasKey(c => new { c.Id, c.PartitionKey1, c.PartitionKey2, c.PartitionKey3 });
+                    cb.HasPartitionKey(
```
Comment (roji) [inline test/EFCore.Cosmos.FunctionalTests/EndToEndCosmosTest.cs:1527]: Am sad about these (doesn't seem like they reach our line char limit) but whatever

File: test/EFCore.Cosmos.FunctionalTests/Query/InheritanceQueryCosmosFixture.cs [hunk 2]
```diff
@@ -36,10 +37,13 @@ protected override void OnModelCreating(ModelBuilder modelBuilder, DbContext con
         modelBuilder.Entity<Plant>().ToContainer("Plants");
         modelBuilder.Entity<Plant>().Property<string>("Discriminator").ToJsonProperty("_type");
         modelBuilder.Entity<Country>().ToContainer("Countries");
-        modelBuilder.Entity<Drink>().ToContainer("Drinks");;
+        modelBuilder.Entity<Drink>().ToContainer("Drinks");
+        ;
```
Comment (roji) [inline test/EFCore.Cosmos.FunctionalTests/Query/InheritanceQueryCosmosFixture.cs:41]:
```suggestion
```

File: test/EFCore.Cosmos.FunctionalTests/Query/InheritanceQueryCosmosFixture.cs [hunk 3]
```diff
@@ -36,10 +37,13 @@ protected override void OnModelCreating(ModelBuilder modelBuilder, DbContext con
...
         modelBuilder.Entity<KiwiQuery>().ToContainer("Animals");
         modelBuilder.Entity<AnimalQuery>().ToContainer("Animals");
-        modelBuilder.Entity<BirdQuery>().ToContainer("Animals");;
-        modelBuilder.Entity<KiwiQuery>().ToContainer("Animals");;
+        modelBuilder.Entity<BirdQuery>().ToContainer("Animals");
+        ;
+        modelBuilder.Entity<KiwiQuery>().ToContainer("Animals");
+        ;
```
Comment (roji) [inline test/EFCore.Cosmos.FunctionalTests/Query/InheritanceQueryCosmosFixture.cs:47]:
```suggestion
        modelBuilder.Entity<KiwiQuery>().ToContainer("Animals");
```

File: test/EFCore.Cosmos.FunctionalTests/Query/ReadItemPartitionKeyQueryFixtureBase.cs [hunk 4]
```diff
@@ -371,28 +371,28 @@ private static List<SinglePartitionKeyEntity> CreateSinglePartitionKeyEntities()
         private static List<OnlyHierarchicalPartitionKeyEntity> CreateOnlyHierarchicalPartitionKeyEntities()
             => new()
             {
-                new()
+                new OnlyHierarchicalPartitionKeyEntity
```
Comment (roji) [inline test/EFCore.Cosmos.FunctionalTests/Query/ReadItemPartitionKeyQueryFixtureBase.cs:374]: My preference on these would be to explicitly specify the list type (`new List<OnlyHierarchyPartitionKeyEntity>()` just above) and then use target-typed new in the elements - hopefully the editorconfig setting allows for that... But no need to block this PR for that.

File: test/EFCore.Cosmos.FunctionalTests/VectorSearchCosmosTest.cs [hunk 5]
```diff
@@ -214,7 +214,7 @@ await Assert.ThrowsAsync<InvalidOperationException>(
         // Assert.Equal(3, booksFromStore.Count);
 
         AssertSql(
-);
+        );
```
Comment (roji) [inline test/EFCore.Cosmos.FunctionalTests/VectorSearchCosmosTest.cs:217]: Move to line above

File: test/EFCore.Cosmos.FunctionalTests/VectorSearchCosmosTest.cs [hunk 6]
```diff
@@ -384,7 +384,7 @@ protected override Task SeedAsync(PoolableDbContext context)
                     NestedOwned = new Owned2 { Prop = "7" },
                     NestedOwnedCollection = new List<Owned2> { new() { Prop = "71" }, new() { Prop = "72" } }
                 },
-                OwnedCollection = new List<Owned1> { new Owned1 { Prop = 71 }, new Owned1 { Prop = 72 } }
+                OwnedCollection = new List<Owned1> { new() { Prop = 71 }, new() { Prop = 72 } }
```
Comment (roji) [inline test/EFCore.Cosmos.FunctionalTests/VectorSearchCosmosTest.cs:387]: With regards to my comment above on target-typed, this is how I think it should be - explicitly type the list, implicitly the elements in the initializer...

**Review by roji — APPROVED** [review]

File: test/EFCore.Cosmos.FunctionalTests/EndToEndCosmosTest.cs (hunk 1 above)
Comment (ajcvickers) [reply test/EFCore.Cosmos.FunctionalTests/EndToEndCosmosTest.cs:1527]: Our rules say we always expand these. Not a fan of it either.

File: test/EFCore.Cosmos.FunctionalTests/VectorSearchCosmosTest.cs (hunk 5 above)
Comment (ajcvickers) [reply test/EFCore.Cosmos.FunctionalTests/VectorSearchCosmosTest.cs:217]: This is fixed in the other PR.

**Review by AndriySvyryd — APPROVED** [review]

**Pull request merged**